src/drivers/fs/virtio_fs.rs

```diff
@@ -153,17 +153,21 @@ impl VirtioFsDriver {
 }
 
 impl FuseInterface for VirtioFsDriver {
-	fn send_command<S, T>(&mut self, cmd: &fuse::Cmd<S>, rsp: &mut fuse::Rsp<T>)
-	where
-		S: fuse::FuseIn + core::fmt::Debug,
-		T: fuse::FuseOut + core::fmt::Debug,
+	fn send_command<const CODE: u32>(
+		&mut self,
+		cmd: &<fuse::Op<CODE> as fuse::OpTrait>::Cmd,
+		rsp: &mut <fuse::Op<CODE> as fuse::OpTrait>::Rsp,
+	) where
+		fuse::Op<CODE>: fuse::OpTrait,
 	{
 		if let Some(mut buff_tkn) = self.ready_queue.pop() {
 			let cmd_len = Some(cmd.len());
 			let rsp_len = Some(rsp.len());
 			buff_tkn.restr_size(cmd_len, rsp_len).unwrap();
 
-			let transfer_tkn = buff_tkn.write(Some(cmd), Some(rsp)).unwrap();
+			let transfer_tkn = buff_tkn
+				.write(Some(cmd), None::<&<fuse::Op<CODE> as fuse::OpTrait>::Rsp>)
+				.unwrap();
```

Comment:
Did you change `Some(rsp)` to `None` purposefully?

Reply:
Yes. As far as I understand, FUSE does not need the memory allocated for the response to be initialized at all. It reads the size of the writable section from the command struct. `BufferToken::write` copies the the argument when it is Some and since we don't initialize the response, I bypassed the copying.

Reply:
The part of the request structure that corresponds to our `Rsp` is [referred](https://docs.oasis-open.org/virtio/virtio/v1.2/cs01/virtio-v1.2-cs01.html#x1-4670001) to as a “[d]evice-writable part” in the file system device section of the Virtio specification and the Virtqueue section [specifies](https://docs.oasis-open.org/virtio/virtio/v1.2/virtio-v1.2.html#x1-440001) that “a device SHOULD NOT read a device-writable buffer (it MAY do so for debugging or diagnostic purposes).”

Reply:
Since this is true for Virtqueue's in general, I think we can also remove receive buffer copying from the Virtqueue interface (unless we want to keep it for “debugging or diagnostic purposes,” as mentioned in the specification), though it should probably be a separate PR since it touches parts of the code base outside FUSE.

Reply:
All right, sounds good! 👍